nodejs/awsx/lb/loadBalancer.ts

```diff
@@ -30,6 +30,13 @@ export abstract class LoadBalancer extends pulumi.ComponentResource {
     constructor(type: string, name: string, args: LoadBalancerArgs, opts: pulumi.ComponentResourceOptions) {
         super(type, name, {}, opts);
 
+        if (args.loadBalancer) {
+            this.loadBalancer = args.loadBalancer;
+            this.vpc = x.ec2.Vpc.fromExistingIds(`${name}-vpc`, { vpcId: this.loadBalancer.vpcId });
+            this.securityGroups = x.ec2.getSecurityGroups(this.vpc, name, this.loadBalancer.securityGroups.get(), { parent: this }) || [];
```

Comment:
I don't think `.get()` will work here since the `Output` won't necessarily be known at this point. `securityGroups` also only applies to ALBs, so I don't think you can indiscriminately attempt to get and set like this. Doing an `apply` against `loadBalancer.securityGroups` would be unideal as we'd end up with resource creation within that `apply`, so I'm not sure of a good way around this other than to require passing in the security groups.

Reply:
Updated. Thanks so much for taking a look at this and for fixing the issue in the aws repo as well